src/cursor/cur_std.c

```diff
@@ -970,37 +1003,6 @@ __cursor_reuse_or_init(WT_SESSION_IMPL *session, WT_CURSOR *cursor, const char *
     if (*readonlyp) /* We do not cache read-only cursors. */
         F_CLR(cursor, WT_CURSTD_CACHEABLE);
 
-    /*
-     * If an index cursor is opened with the dump configuration option, then this function is called
-     * on the index files, with the dump config string, and with the index cursor as an owner. We
-     * don't want to create a dump cursor in that case, because we'll create the dump cursor on the
-     * index cursor itself.
-     */
-    WT_RET(__wt_config_gets_def(session, cfg, "dump", 0, &cval));
-    owner = ownerp ? *ownerp : NULL;
-    if (cval.len != 0 && owner == NULL) {
-        uint64_t dump_flag;
-        if (WT_CONFIG_LIT_MATCH("json", cval))
-            dump_flag = WT_CURSTD_DUMP_JSON;
-        else if (WT_CONFIG_LIT_MATCH("print", cval))
-            dump_flag = WT_CURSTD_DUMP_PRINT;
-        else if (WT_CONFIG_LIT_MATCH("pretty", cval))
-            dump_flag = WT_CURSTD_DUMP_PRETTY;
-        else if (WT_CONFIG_LIT_MATCH("pretty_hex", cval))
-            dump_flag = WT_CURSTD_DUMP_HEX | WT_CURSTD_DUMP_PRETTY;
-        else
-            dump_flag = WT_CURSTD_DUMP_HEX;
-        F_SET(cursor, dump_flag);
-
-        /*
-         * Dump cursors should not have owners: only the top-level cursor should be wrapped in a
-         * dump cursor.
-         */
-        WT_RET(__wti_curdump_create(cursor, owner, &cdump));
-        *ownerp = cdump;
-        F_CLR(cursor, WT_CURSTD_CACHEABLE);
-    } else
-        cdump = NULL;
     *cdumpp = cdump;
```

Comment:
While I know this is out of the conditional that you moved, it feels weird to leave this line now far away from the dump related code. Can you consider moving it up too?

Reply:
Yes, moved up closer.